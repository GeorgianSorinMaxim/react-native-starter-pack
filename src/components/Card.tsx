import React from "react";
import {
  Image,
  Linking,
  StyleSheet,
  TouchableOpacity,
  View,
} from "react-native";

import ImagePlaceholder from "./ImagePlaceholder";

import Colors from "../constants/Colors";
import BodyText from "./BodyText";

import { University } from "../store/reducers/data";

interface Props {
  item: University;
}

export const Card = ({ item }: Props) => {
  const onItemTap = (url: string) => {
    Linking.openURL(url).catch(err => console.warn(err));
  };

  const { imageUrl, geo } = item;

  let address = "No address";
  if (geo && geo.address) {
    const { streetAddress, postalCode, addressLocality } = geo.address;
    address = `${streetAddress}, ${postalCode}, ${addressLocality}`;
  }

  return (
    <TouchableOpacity onPress={() => onItemTap(item.url)}>
      <View style={styles.row}>
        <View style={styles.leftColumn}>
          <BodyText style={styles.universityName}>{item.name}</BodyText>
          <BodyText style={styles.address}>{address}</BodyText>
        </View>
        <View style={styles.rightColumn}>
          {!imageUrl || imageUrl.includes("svg") ? (
            <ImagePlaceholder />
          ) : (
            <Image source={{ uri: imageUrl }} style={styles.image} />
          )}
        </View>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  row: {
    marginBottom: 12,
    borderRadius: 6,
    width: "100%",
    padding: 12,
    display: "flex",
    alignItems: "flex-start",
    borderColor: Colors.grey,
    borderWidth: 1,
    flexDirection: "row",
    justifyContent: "space-between",
  },
  leftColumn: {
    width: "60%",
  },
  rightColumn: {
    width: "40%",
    alignItems: "flex-end",
  },
  universityName: {
    marginTop: 12,
    fontWeight: "bold",
    color: Colors.black,
    overflow: "hidden",
  },
  address: {
    color: Colors.black,
    overflow: "hidden",
    paddingTop: 12,
  },
  image: {
    width: 100,
    height: 100,
    resizeMode: "contain",
  },
});

export default Card;
