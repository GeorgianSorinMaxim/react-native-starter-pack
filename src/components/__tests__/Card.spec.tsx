import * as React from "react";
import renderer from "react-test-renderer";

import Card from "../Card";

it(`renders correctly`, () => {
  const defaultProps = {
    item: {
      name: "Nando's",
      url: "https://www.nandos.com",
      geo: {
        address: {
          streetAddress: "",
          postalCode: "",
          addressLocality: ""
        }
      }
    }
  };
  const tree = renderer.create(<Card {...defaultProps} />).toJSON();

  expect(tree).toMatchSnapshot();
});
