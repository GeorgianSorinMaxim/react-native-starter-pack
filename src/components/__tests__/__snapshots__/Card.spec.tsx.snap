// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "borderColor": "#D4AF37",
        "borderRadius": 6,
        "borderWidth": 1,
        "display": "flex",
        "marginBottom": 12,
        "padding": 12,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#f7f7f7",
          "flexDirection": "row",
          "height": 150,
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 60,
            },
            Object {
              "color": "#BABABA",
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        Ôçê
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#D4AF37",
            "fontWeight": "bold",
            "marginTop": 12,
            "overflow": "hidden",
          },
        ]
      }
    >
      Google.com
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "overflow": "hidden",
            "paddingTop": 12,
          },
        ]
      }
    >
      , , 
    </Text>
  </View>
</View>
`;
